solr/contrib/ltr/example/README.md

```diff
@@ -41,20 +43,84 @@ hard drive|6H500F0|0.330082034|CLICK_LOGS
+
+  1. the user query;
+
+  2. a unique id for a document in the response;
+
+  3. the a score representing the relevance of that document (not necessarily between zero and one);
+
+  4. the source, i.e., if the training record was produced by using interaction data (`CLICK_LOGS`) or by human judgements (`HUMAN_JUDGEMENT`).
+
+## How to produce training data
+
+You might collect data for use with your machine learning algorithm relying on:
+
+  * **Users Interactions**: given a specific query, it is possible to log all the users interactions (e.g., clicks, shares on social networks, send by email etc), and then use them as proxies for relevance;
+  * **Human Judgments**: A training dataset is produced by explitely asking some judges to evaluate the relavance of a document given the query.
+
+### How to prepare training data from interaction data?
+
+There are many ways of preparing interaction data for training a model, and it is outside
+the scope of this readme to provide a complete review of all the techniques.
+In the following we illustrate a simple way for obtaining training
+data from simple interaction data.
+
+Simple interaction data will be a Solr log file, with two different types for record:
+
+  * **query**: when a user performs a query we have a record with `user-id, query, responses`,
+  where `responses` is a list of unique document ids returned for a query.
+
+**Example:**
+
+```
+query: diego, ipod, [SP2514N,6H500F0,F8V7067-APL-KIT,IW-02]
+```
+
+  * **click**: when a user performs a click we have a record with `user-id, document-id, click`
+
+**Example:**
+```
+click: diego,   P2515N
+click: michael, P215N
+click: joshua,  IW-02`.
+```
+
+Given a log composed by records like these, a simple way to produce a training dataset is to group on the query field
+and then assign to each query a relevance score equals to the number of clicks:
+
+```
+hard drive|SP2514N|2|CLICK_LOGS
+hard drive|IW-02|1|CLICK_LOGS
+hard drive|6H500F0|0|CLICK_LOGS
+hard drive|F8V7067-APL-KIT|0|CLICK_LOGS
+```
+
+This is a really trival way to generate a training dataset, and in many settings it will not work.
+Indeed, it is a well known fact that clicks are *biased*: users tend to click  on the first
+results proposed for a query, also if is not relevant. A click on a document in position
+five should be considered more important than a click on a document in position one, because
+the user took the effort to browse the results list until position five.
+Some approaches take into account the time spent on the clicked document (if the user
+spent only two seconds on the document and then clicked on other documents in the list,
+probably she did not intend to click that document).
+
+There are many papers proposing techniques for removing the bias, or for taking into account the click positions,
+a good survey is  [Click Models for Web Search](http://clickmodels.weebly.com/uploads/5/2/2/5/52257029/mc2015-clickmodels.pdf),
```

Comment:
I removed it because I'm afraid it's too much academic and it could confuse readers..